### PR TITLE
Add details to newman types

### PR DESCRIPTION
**Pull request by Graham42**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://github.com/postmanlabs/newman/blob/release/3.x/README.md#api-reference
  - https://github.com/postmanlabs/newman/blob/release/3.x/lib/run/summary.js#L48
  - https://github.com/postmanlabs/newman/blob/release/3.x/lib/run/summary.js#L274
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Graham42** [comment]
> Don't use a 'package.json' for @types dependencies.
This error message doesn't give me any help for what the proper thing to do is, any ideas?

**Review by LogvinovLeon — APPROVED** [review]
Thanks a lot!

File: types/newman/index.d.ts [hunk 1]
```diff
@@ -1,14 +1,161 @@
...
+     * "folder" allows you to skip the entire collection run in case an invalid
+     * folder was specified using the `folder` option or an error was
+     * encountered in general.
+     *
+     * "failure" would gracefully stop a collection run after completing the
+     * current test script.
+     *
+     * Default value: false
+     */
+    bail?: boolean | ["folder"] | ["failure"];
```
Comment (amcasey) [inline types/newman/index.d.ts]: Is the difference between `"folder"` and `["folder"]` important in this case?

File: types/newman/index.d.ts [hunk 2]
```diff
@@ -23,4 +170,12 @@ export interface NewmanRunExecutionAssertion {
...
 }
-export function run(options: any, callback?: (err: Error | null, summary: NewmanRunSummary) => void): void;
+export interface NewmanRunFailure {
+    error: NewmanRunExecutionAssertionError;
+    /** The event where the failure occurred */
+    at: string;
+}
+export function run(
+    options: NewmanRunOptions,
+    callback?: (err: Error | null, summary: NewmanRunSummary) => void
```
Comment (amcasey) [inline types/newman/index.d.ts]: Are `Error` and `NewmanRunExecutionAssertionError` (supposed to be) related?

File: types/newman/index.d.ts (hunk 1 above)
Comment (Graham42) [reply types/newman/index.d.ts]: Yup https://github.com/postmanlabs/newman/blob/release/3.x/lib/run/index.js#L101

Comment (amcasey) [reply types/newman/index.d.ts]: Thanks for confirming!

File: types/newman/index.d.ts (hunk 2 above)
Comment (Graham42) [reply types/newman/index.d.ts]: They're different from what I could tell. This callback `Error` is for fatal newman errors but `NewmanRunExecutionAssertionError` is for test / script failures

callback error:
https://github.com/postmanlabs/newman/tree/release/3.x#newmanruncallbackerror-object--summary-object
> By default, only fatal errors, such as the ones caused by any fault inside Newman is passed on to this argument.

execution error:
https://github.com/postmanlabs/newman/blob/release/3.x/lib/run/summary.js#L240
seem to have errors that could be a syntax error, assertion error, or other error.

Comment (amcasey) [reply types/newman/index.d.ts]: Thanks for confirming!

**Comment by Graham42** [comment]
Thanks @LogvinovLeon  and @amcasey  for the review :smile:

**Comment by Graham42** [comment]
@amcasey I couldn't find this documented anywhere... When can I expect this package to be published?

**Comment by LogvinovLeon** [comment]
@Graham42 https://github.com/DefinitelyTyped/DefinitelyTyped#my-pr-is-merged-when-will-the-types-npm-package-be-updated

NPM packages should update within a few hours. If it's been more than 24 hours, ping RyanCavanaugh and andy-ms on the PR to investigate.

**Comment by Graham42** [comment]
Thanks, I missed that.
ping @RyanCavanaugh @andy-ms

**Comment by Graham42** [comment]
ping @RyanCavanaugh @andy-ms

**Pull request merged**